internal/fs/inode/dir.go

```diff
@@ -691,10 +691,10 @@ func (d *dirInode) readObjects(
 		}
 	}
 
-	// Return an appropriate continuation token, if any.
+	// return an appropriate continuation token, if any.
 	newTok = listing.ContinuationToken
 
-	if !d.implicitDirs {
+	if !d.implicitDirs && !d.isHNSEnabled {
```

Comment:
Can you please use `!d.isBucketHierarchical() `

Reply:
yes actually d.isBucketHierarchical() suffices both.. Tulsi added it later so I think its redundant at older places and places where I updated the code... yes agreed we can try and remove as when we touch the check code again in any new changes

Reply:
Okk. Pls add a todo in code as a comment otherwise its easy to miss them if you are not doing it now.